src/sentry/workflow_engine/handlers/action/notification/handler.py

```diff
@@ -203,14 +207,26 @@ def handle_workflow_action(job: WorkflowJob, action: Action, detector: Detector)
             raise NotificationHandlerException(e)
 
     @staticmethod
-    def target(action: Action) -> RpcUser | Team | str | None:
+    def target(action: Action) -> OrganizationMember | Team | str | None:
         target_identifier = action.config.get("target_identifier")
         if target_identifier is None:
             return None
 
+        try:
+            dcga = DataConditionGroupAction.objects.get(action=action)
+        except DataConditionGroupAction.DoesNotExist:
+            return None
+        try:
+            dcg = DataConditionGroup.objects.get(id=dcga.condition_group_id)
+        except DataConditionGroup.DoesNotExist:
+            return None
+
```

Comment:
We could do this in one query, since all we need is the organization here. I think we also should do this inside the `if target_type == ActionTarget.USER.value:` if statement so that we don't make unnecessary queries